Mage/src/main/java/mage/abilities/common/BecomesTappedOneOrMoreTriggeredAbility.java

```diff
@@ -37,12 +42,21 @@ public boolean checkEventType(GameEvent event, Game game) {
     }
 
     @Override
-    public boolean checkTrigger(GameEvent event, Game game) {
-        TappedBatchEvent batchEvent = (TappedBatchEvent) event;
-        return batchEvent
-                .getTargetIds()
+    public Stream<TappedEvent> filterBatchEvent(GameEvent event, Game game) {
+        return ((TappedBatchEvent) event)
+                .getEvents()
                 .stream()
-                .map(game::getPermanent)
-                .anyMatch(p -> filter.match(p, getControllerId(), this, game));
+                .filter(e -> Optional
+                        .of(e)
+                        .map(TappedEvent::getTargetId)
+                        .map(game::getPermanent)
+                        .filter(p -> filter.match(p, getControllerId(), this, game))
+                        .isPresent()
+                );
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return filterBatchEvent(event, game).findAny().isPresent();
```

Comment:
See above about List. Example:
* `BatchTriggeredAbility` extends `TriggeredAbilityImpl`;
  * it's has default method `filterEvent` with `return true`;
  * it's has final method like `List findFilteredEvents(Event, Game)` that calls implementations of `filterEvent`;
  * it's has default implementation of `checkTrigger` that `return !findFilteredEvents(event, game).isEmpty`;
* All implementations of `BatchTriggeredAbility`:
  *  overrides `filterEvent` method to filter only needed;
  *  override `checkTrigger` method to setup dynamic targets or other additional code.

That's all will work same way as other triggered abilities, but with additional middleware (filterEvent);

P.S. I don't like generic usages all around. If it possible to use standard classes then it will be better (extends basic ability, extends basic event, etc). If it (generic) can simplify some code then can be keep.

Reply:
Well I won't do all that additional work that doesn't accomplish much.
So either someone else take the PR from here, or we can close it and leave the bugs.

Reply:
I guess I'll work on that then.